### ISSUE TITLE
Error in requestSubscription (iOS)

### ISSUE THREAD

**Issue by igorjos** [dooboolab/react-native-iap#2267]
**Description**

Trying to make a new request for subscription on IOS, but it returns the following error:
` LOG  [Error: Invalid product ID. Did you call getProducts/Subscriptions]`

While before I call the `requestSubscription({sku: 'Yearly_Subscription'})` method I'm getting a list of products:

```javascript
[{"currency": "", "description": "APP monhtly", "localizedPrice": "$0.99", "platform": "ios", "price": "0.99", "productId": "Monthly_Subscription", "subscriptionPeriodNumberIOS": "undefined", "subscriptionPeriodUnitIOS": undefined, "title": "Monthly Subscription", "type": "subs"}, {"currency": "", "description": "App annual", "localizedPrice": "$9.99", "platform": "ios", "price": "9.99", "productId": "Yearly_Subscription", "subscriptionPeriodNumberIOS": "undefined", "subscriptionPeriodUnitIOS": undefined, "title": "Annual Subscription", "type": "subs"}]
```

**Environment:**

- react-native-iap: 12.7.4
- react-native: 0.70.6
- Platforms (iOS, simulator):

**Comment by cnickless** [comment]
Having the same issue/error today using Sk2.  Can confirm a valid sku is passed after retrieving from getSubscriptions, but always receive an error that it's an Invalid product ID.

**Environment:**

- react-native-iap: 12.7.4
- react-native: 0.70.4
- Platforms (iOS, iPhone 6S, iPhone 13)

**Comment by cnickless** [comment]
@igorjos Wow. You saved me! In my case, I had wrapped my app with an outer <Subscription> layer that has all necessary IAP listeners configured. This is where I would initConnection and endConnection for the session (open/close of app). I would also fetch subscription products at this level and store them in context to be referenced later from the purchase screen.

I've now moved getSubscriptions to fire just before requestSubscription. The back to back calls have "fixed" this issue.

Cheers!
